packages/emf/backend/sirius-components-emf/src/main/java/org/eclipse/sirius/components/emf/services/JSONResourceFactory.java

```diff
@@ -21,18 +21,43 @@
 import org.eclipse.sirius.emfjson.resource.JsonResourceImpl;
 
 /**
- * Special factory for {@link JsonResource} for Sirius Web.</br>
+ * Special factory for {@link JsonResource}.</br>
  * This factory ensures that objects in that resource are identified by their uuid.
  *
  * @author lfasani
  */
-public class SiriusWebJSONResourceFactoryImpl extends ResourceFactoryImpl {
+public class JSONResourceFactory extends ResourceFactoryImpl {
 
     @Override
     public JsonResource createResource(URI uri) {
         Map<String, Object> options = new HashMap<>();
 
         options.put(JsonResource.OPTION_ID_MANAGER, new EObjectIDManager());
+        options.put(JsonResource.OPTION_DISPLAY_DYNAMIC_INSTANCES, true);
+
         return new JsonResourceImpl(uri, options);
     }
+
+    /**
+     * Create a resource which URI is made of a correct scheme and the given id as path.
+     *
+     * @param resourceId
+     *            the id of the resource
+     */
+    public JsonResource createResource(String resourceId) {
```

Comment:
The naming is confusing. It's the same name (`createResource`) for the version which takes a URI and the version which takes a "raw" resourceId.
It's pretty easy to use the wrong one. For example in `DeleteDocumentEventHandlerTests` we do:

```java
new JSONResourceFactory().createResource(URI.createURI(document.getId().toString()));
```

which creates a Resource with the wrong URI (`uri='a171a541-44cc-4607-9583-44fb6524e481'`). It should be:

```java
new JSONResourceFactory().createResource(document.getId().toString());
```

which looks very similar but behave very differently.

I wonder if `createResource(URI uri)` should just throw an `IllegalArgumentException` with a proper message if given a URI without a scheme. It would make most latent bugs obvious and identify the problematic caller immediatly.

Reply:
I renamed `createResource(String)` to `createResourceFromPath`